### ISSUE TITLE
Are these methods suitable for semantic segmentation tasks and 3d images?

### ISSUE THREAD

**Issue by Willzy-x** [IBM/adversarial-robustness-toolbox#220]
Hi,
 
I noticed that the attack methods are always initialized with a trained classifier. Does it mean that those methods can only be implemented on classification tasks?
In addition, I also want to know whether these methods, for example, Gaussian Data Augmentation in defence methods, can be applied to 3d images(medical images) as well.

Best,
Yucheng

**Comment by beat-buesser** [comment]
Hi @Willzy-x Thank you for using ART and raising these interesting questions.

ART is currently tested only for classification tasks and for example `KerasClassifier` only supports classification losses, so far. I have seen people using ART for regression tasks with relatively small modifications to the source code, mainly on internal label handling and loss function definition. It would definitely be interesting to extend ART officially to include regression. Do you have a specific regression task you are working on?

`GaussianAugmentation` should be able to run for any shape as long as the first dimension is the batch size. We should update the docstring for parameter `x` accordingly. Therefore I think `GaussianAugmentation` can be used for 3d medical images.

**Comment by Willzy-x** [comment]
Nope, I am not working on a regression task. Instead, I am working on a semantic segmentation task on medical images, which can be understood as classifying every single pixel on an image(of course, voxel on a 3d image).

And I hope that some methods in adversarial machine learning will improve my network's performance or robustness, such as adversarial training, virtual adversarial training and so on.

**Comment by beat-buesser** [comment]
Ok, I see, because you asked if there is only support for classification I automatically assumed you are looking at regression. I think pixel-wise classification should be straightforward with ART.

**Comment by Willzy-x** [comment]
Ok, Thanks for the help. 😄
